src/main/java/com/google/devtools/build/lib/skyframe/DependencyResolver.java

```diff
@@ -497,80 +499,97 @@ public static DependencyContext getDependencyContext(
   private void handleException(ExtendedEventHandler listener, Target target, Exception untyped)
       throws ReportedException {
 
-    if (untyped instanceof DependencyEvaluationException e) {
-      String errorMessage = e.getMessage();
-      if (!e.depReportedOwnError()) {
-        listener.handle(Event.error(e.getLocation(), e.getMessage()));
-      }
+    throw switch (untyped) {
```

Comment:
This is a useful cleanup: is it related to the new handling for `exec_group_compatible_with`? Can it be split into a separate PR to make review (and future commit archaeology) easier?